pydra/engine/core.py

```diff
@@ -916,7 +926,10 @@ async def _run(self, submitter=None, rerun=False, **kwargs):
             try:
                 self.audit.monitor()
                 await self._run_task(submitter, rerun=rerun)
-                result.output = self._collect_outputs()
+                result.output, err = self._collect_outputs()
```

Comment:
why would/should collect_outputs return an error? and not _run_task raise an exception?

Reply:
or is this specifically being done for workflows? if so can't _collect_outputs also raise an exception? the exception is already being trapped.